ops/services/app_service/metabase/main.tf

```diff
@@ -1,3 +1,73 @@
+locals {
+  app_setting_defaults = {
+    "MB_DB_CONNECTION_URI"           = var.postgres_url
+    "WEBSITE_VNET_ROUTE_ALL"         = 1
+    "WEBSITE_DNS_SERVER"             = "168.63.129.16"
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = var.ai_instrumentation_key
+  }
+
+  grant_command = <<EOF
+  psql \
+  --host ${var.postgres_server_name} \
+  --port ${var.postgres_port} \
+  --username ${var.administrator_login} \
+  --dbname "${var.postgres_db_name}" \
+  --command "
+    CREATE ROLE ${data.azurerm_key_vault_secret.postgres_readonly_user.value} with LOGIN ENCRYPTED PASSWORD
+    '${data.azurerm_key_vault_secret.postgres_readonly_pass.value}';
+    GRANT USAGE ON SCHEMA public TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility_device_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
```

Comment:
Two or three related worries:
* do we need to do a `revoke` first to fix the public schema?
* should we parameterize this with the schema name so we don't have to go back and fix if we move out of `public`?
* what happens if we add a new table?

Reply:
Addressed point by point for readability.

- yep, we probably do need a revoke first. I've added `REVOKE ALL PRIVILEGES ON SCHEMA public FROM PUBLIC;` but let me know if thats not what you're looking for
- I parameterized the schema name to locals since it didn't seem sensitive, but let me know if you want me to move that elsewhere.
- from my understanding of Postgres permissions (which is not the best) the user wouldn't be able to see the new table at all unless access is explicitly granted.

Reply:
That is the default, yeah. I'm wondering if there's a way to finesse it, or in the alternative to systematize dealing with it. Worth a little thought, will ping you if I have something approaching a constructive suggestion.

Reply:
Discussed offline - there should be a way to move some of the permissions into liquibase, so that when adding a new table in the changelog you have to specify which users will have permissions on it before submitting. However, I'd prefer to do that in a separate PR so that this one can focus on the initial user creation/metabase setup, is that alright?

Reply:
Was just this minute suggesting literally that same solution in a different offline chat. Should probably get us all three into a single chat instead of playing telephone, sorry!